lua/toggleterm/ui.lua

```diff
@@ -183,13 +183,27 @@ local split_commands = {
   },
 }
 
+--- @param size number/function
+local function resolve_size(size)
+  if type(size) == 'number' then
+    return function()
+      return size
+    end
+  end
+  if type(size) == 'function' then
+    return size
+  end
+  error(string.format('The input %s is not of type "number" or "function".',size))
```

Comment:
I try to avoid the use of `error` since it tends to print a pretty ugly message + stack trace to the message area I think it's nicer to print the message using `utils.echoerr` which spares the user details they likely don't care about